src/cript/api/api.py

```diff
@@ -38,52 +38,20 @@ def _get_global_cached_api():
     return _global_cached_api
 
 
-def _prepare_host(host: str) -> str:
-    """
-    prepares the host and gets it ready to be used within the api client
-
-    1. removes any trailing spaces from the left or right side
-    1. removes "/" from the end so that it is always uniform
-    1. adds "/api", so all queries are sent directly to the API
-
-    Parameters
-    ----------
-    host: str
-        api host
-
-    Returns
-    -------
-    host: str
-    """
-    # strip any empty spaces on left or right
-    host = host.strip()
-
-    # strip ending slash to make host always uniform
-    host = host.rstrip("/")
-
-    host = f"{host}/api/v1"
-
-    # if host is using unsafe "http://" then give a warning
-    if host.startswith("http://"):
-        warnings.warn("HTTP is an unsafe protocol please consider using HTTPS.")
-
-    if not host.startswith("http"):
-        raise InvalidHostError("The host must start with http or https")
-
-    return host
-
-
 class API:
     """
     ## Definition
     API Client class to communicate with the CRIPT API
     """
 
-    _host: str = ""
+    _raw_host: str = ""
```

Comment:
how come we are keeping this?
do we need this anywhere else?

Reply:
we are keeping it now, because it is so trivial to construct the host from it.
It might be helpful to have for debugging at some point. (You actually know what the user submitted, and something goes wrong with constructing the url from it.)

Reply:
this makes it confusing for anyone reading the code because there is 2 different host and it's not needed, debugging the string is going to be pretty easy, I can't think of a scenario when this is needed

Reply:
No, there is only one host.
Especially user facing there is only one host.

I don't know you asked me to figure out a better way of handling this.
This is my solution. Hard coding the api modifier was a bad solution, so we needed something better.